beginner_source/nlp/word_embeddings_tutorial.py

```diff
@@ -1,158 +1,135 @@
 
-where the 1 is in a location unique to :math:`w`. Any other word will
-have a 1 in some other location, and a 0 everywhere else.
+여기서 1은 해당 벡터가 표현하고자 하는 단어에 해당하는 위치 1곳에 자리합니다. (나머지는 전부
+0입니다.) 다른 단어를 나타내는 벡터에선 1이 다른 곳에 위치해 있겠죠.
 
-There is an enormous drawback to this representation, besides just how
-huge it is. It basically treats all words as independent entities with
-no relation to each other. What we really want is some notion of
-*similarity* between words. Why? Let's see an example.
+원핫 인코딩은 만들기가 쉽다는 장점이 있지만, 단순한 만큼 단점도 있습니다. 일단 단어 벡터 한 개는
+모든 단어를 표현할 수 있을 만한 크기가 되어야 합니다. 우리가 얼마나 많은 종류의 단어를
+사용하는지를 생각 한다면 어마어마하게 큰 벡터라는 것을 알 수 있죠. 이 뿐만이 아닙니다.
+원핫 벡터는 모든 단어를 독립적인 개체로 가정하는 것을 볼 수 있습니다. 즉, 공간상에서
+완전히 다른 축에 위치해 있어서 단어간의 관계를 나타낼 수가 없습니다. 하지만 우리는 단어
+사이의 *유사도*를 어떻게든 계산하고 싶은거죠. 왜 유사도가 중요하냐구요? 다음 예제를 봅시다.
 
-Suppose we are building a language model. Suppose we have seen the
-sentences
+우리의 목표가 언어 모델을 만드는 것이라고 가정하고 다음의 문장이 학습 데이터로써 주어졌다고 해봅시다.
 
-* The mathematician ran to the store.
-* The physicist ran to the store.
-* The mathematician solved the open problem.
+* 수학자가 가게로 뛰어갔다.
+* 물리학자가 가게로 뛰어갔다.
+* 수학자가 리만 가설을 증명했다.
 
-in our training data. Now suppose we get a new sentence never before
-seen in our training data:
+또한 학습 데이터에는 없는 아래 문장이 있다고 생각해봅시다.
 
-* The physicist solved the open problem.
+* 물리학자가 리만 가설을 증명했다.
 
-Our language model might do OK on this sentence, but wouldn't it be much
-better if we could use the following two facts:
+ASCII 코드나 원핫 인코딩 기반 언어 모델은 위 문장을 어느정도 다룰 수 있겠지만, 개선의 여지가 있지 않을까요?
+먼저 아래의 두 사실을 생각해봅시다.
 
-* We have seen  mathematician and physicist in the same role in a sentence. Somehow they
-  have a semantic relation.
-* We have seen mathematician in the same role  in this new unseen sentence
-  as we are now seeing physicist.
+* '수학자'와 '물리학자'가 문장 내에서 같은 역할을 맡고 있습니다. 이 두 단어는 어떻게든 의미적인 연관성이 있을 겁니다.
+* 새로운 문장에서 '물리학자'가 맡은 역할을 '수학자'가 맡는 것을 학습 데이터에서 본 적이 있습니다.
 
-and then infer that physicist is actually a good fit in the new unseen
-sentence? This is what we mean by a notion of similarity: we mean
-*semantic similarity*, not simply having similar orthographic
-representations. It is a technique to combat the sparsity of linguistic
-data, by connecting the dots between what we have seen and what we
-haven't. This example of course relies on a fundamental linguistic
-assumption: that words appearing in similar contexts are related to each
-other semantically. This is called the `distributional
-hypothesis <https://en.wikipedia.org/wiki/Distributional_semantics>`__.
+우리 모델이 위의 사실을 통해 '물리학자'가 새 문장에 잘 들어 맞는다는 것을 추론할 수
+있다면 참 좋을 것입니다. 이것이 위에서 언급한 유사도의 의미입니다. 철자적 유사도 뿐
+아니라 *의미적 유사도*인 것입니다. 이것이야말로 언어 데이터에 내재하는 희박성 (sparsity)에
+대한 처방이 될 것입니다. 우리가 본 것과 아직 보지 않은 것 사이를 이어주는 것이죠.
+앞으로는 다음의 언어학적 기본 명제를 가정하도록 합시다. 바로 비슷한 맥락에서 등장하는
+단어들은 서로 의미적 연관성을 가진다는 것입니다. 언어학적으로는 '분산 의미 가설' 혹은 `distributional
+hypothesis <https://en.wikipedia.org/wiki/Distributional_semantics>`__라고 합니다.
```

Comment:
```suggestion
단어들은 서로 의미적 연관성을 가진다는 것입니다. 언어학적으로는 `분산 의미 가설` 혹은 `distributional
hypothesis <https://en.wikipedia.org/wiki/Distributional_semantics>`__ 라고 합니다.
```

Reply:
또는, 아래와 같은 식도 괜찮을 것 같습니다.
```suggestion
단어들은 서로 의미적 연관성을 가진다는 것입니다. 언어학적으로는 `분산 의미 가설(distributional
hypothesis) <https://en.wikipedia.org/wiki/Distributional_semantics>`__ 이라고도 합니다.
```